### ISSUE TITLE
supported-speed not present for interfaces on OPX VM

### ISSUE THREAD

**Issue by james-jra** [open-switch/opx-cps#58]
On the OPX Virtual Machine, `supported-speed` is not returned for interfaces.
How can I determine which values for speed can be used for interfaces on virtual machines?

**Trace (hardware)**
OPX 2.2.0-dev2 on Dell Z9100 - supported-speed is included:
```
root@OPX:~# sudo cps_get_oid.py dell-base-if-cmn/if/interfaces/interface | grep supported
dell-if/if/interfaces-state/interface/supported-speed = 4
...
dell-if/if/interfaces-state/interface/supported-speed = 3,4,5,6,10,7
dell-if/if/interfaces-state/interface/supported-speed = 3,4,5,6,10,7

root@OPX:~# sudo cps_get_oid.py dell-base-if-cmn/if/interfaces/interface | grep supported
dell-if/if/interfaces-state/interface/supported-speed = 4
...
dell-if/if/interfaces-state/interface/supported-speed = 3,4,5,6,10,7
dell-if/if/interfaces-state/interface/supported-speed = 3,4,5,6,10,7
```

**Trace (VM)**
On OPX 2.2.0-dev2 VM - no supported-speed:
```
opxUser@opx_vm:~$ sudo cps_get_oid.py  dell-base-if-cmn/if/interfaces/interface | grep supported | wc -l
0

opxUser@opx_vm:~$ sudo cps_get_oid.py  dell-base-if-cmn/if/interfaces/interface | grep supported | wc -l
0
```

On OPX 2.1.0 VM - no supported-speed:
```
opxUser@opx_vm:~$ sudo cps_get_oid.py  dell-base-if-cmn/if/interfaces/interface | grep supported | wc -l
0

opxUser@opx_vm:~$ sudo cps_get_oid.py  observed dell-base-if-cmn/if/interfaces-state/interface | grep supported | wc -l
0
```

**Comment by atanu-mandal** [comment]
We had fixed this issue for VM platform sometimes back that was accidentally removed during recent upgrade. We are working on to get those changes back to git. 
Thanks for reporting this.

**Comment by atanu-mandal** [comment]
Fix is now merged
https://review.openswitch.net/#/c/14507/

**Comment by mcallisterjp** [comment]
This is still not working as of the 2.2.0 version on `deb http://deb.openswitch.net oldtesting opx`.

Which version has this been merged in for? I can't see that information on the gerrit page in a way I understand.

**Comment by mcallisterjp** [comment]
Here's the version on `oldtesting` now:
```
root@opx_vm1:/home/opxUser# opx-show-packages 
Name                           Original Version     Current Version      Altered              Description                                                  
================================================================================================================================================================

libopx-nas-l3-1                3.2.0+opx1            3.2.0+opx1          No                    This package contains base layer 3 functionality for the Openswitch software. 

libopx-nas-l2-1                1.17.0+opx1           1.17.0+opx1         No                    This package contains L2 Network abstraction features       

opx-kernel-dell-3.16.0-4-amd64 3.16.43-2+deb8u2      3.16.43-2+deb8u2    No                    Linux 3.16 for 64-bit PCs                                   

opx-platform-config-dell-vm    1.2.1                 1.3.2               No                    Platform specific files and packages for VM simulation of Dell Networking S6000-ON 

libopx-logging1                2.1.0                 2.1.0               No                    Openswitch logging - shared library                         

opx-tools                      1.0.1                 1.0.1               No                    Openswitch tools which will be used in OPX Package Management 

libopx-nas-common1             6.1.0                 6.1.0+opx2          No                    This package contains common utilities for the network abstraction component. 

libopx-common1                 1.4.0                 1.4.0               No                    This package contains general utilities for the Openswitch project. 

opx-tmpctl                     2.1.1                 2.1.1               No                    Files common to all platforms                               

libopx-nas-qos1                2.2.0                 2.2.0               No                    This package contains base QoS functionality for the Openswitch software. 

opx-dell-vm                    1.2.1                 1.3.2               No                    All packages for VM simulation of Dell Networking S6000-ON  

opx-nas-interface              5.10.1+opx1           5.10.1+opx7         No                    This package contains nas-interface for the Openswitch software. 

libopx-pas1                    3.0.1+opx1            3.0.1+opx2          No                    This package contains the Platform Adaptation Service.      

python-opx-logging             2.1.0                 2.1.0               No                    Open Switch logging - python binding                        

libopx-cps1                    3.6.2+opx1            3.6.2+opx1          No                    This package contains the Openswitch object library.  The object library is provides APIs to set/get/delete/create and execute RPC calls using data models provided by components. 

libopx-nas-ndi1                3.26.0                3.26.0              No                    This package contains nas-ndi code                          

libopx-nas-linux1              5.11.0                5.11.0+opx3         No                    This package contains the Linux integration portion of the Network abstractions service. 

opx-nas-l3                     3.2.0+opx1            3.2.0+opx1          No                    This package contains base layer 3 functionality for the Openswitch software. 

opx-nas-l2                     1.17.0+opx1           1.17.0+opx1         No                    This package contains L2 Network abstraction features       

opx-nas-common                 6.1.0                 6.1.0+opx2          No                    This package contains common utilities for the network abstraction component. 

libopx-nas-interface1          5.10.1+opx1           5.10.1+opx7         No                    This package contains nas-interface for the Openswitch software. 

opx-sai-api-dev                32.0.0                32.0.0              No                    This package contains SAI API Module                        

opx-nas-daemon                 2.1.0+opx2            2.1.0+opx2          No                    This package contains the Network Abstraction service.      

opx-core                       2.2.0                 2.2.0               No                    Core (platform-independent) OPX implementation              

libopx-nas-acl1                2.4.0+opx1            2.4.0+opx1          No                    This package contains base ACL functionality for the Openswitch software. 

libopx-sdi-sys-vm1             1.5.1+opx3            1.5.1+opx3          No                    This package contains implementation of System Device Interface(SDI) 

opx-nas-acl                    2.4.0+opx1            2.4.0+opx1          No                    This package contains base ACL scripts for the Openswitch software. 

python-opx-cps                 3.6.2+opx1            3.6.2+opx1          No                    This package contains the Openswitch object library.  The object library is provides APIs to set/get/delete/create and execute RPC calls using data models provided by components. 

libopx-base-model1             3.109.0               3.109.0             No                    Openswitch base model - object models for all services      

opx-alarm                      1.0.2                 1.0.2               No                    Files common to all platforms                               

opx-logging                    2.1.0                 2.1.0               No                    Openswitch logging - logging utilies                        

opx-pas                        3.0.1+opx1            3.0.1+opx2          No                    This package contains the Platform Adaptation Service.      

opx-nas-linux                  5.11.0                5.11.0+opx3         No                    This package contains the Linux integration portion of the Network abstractions service. 

opx-nas-qos                    2.2.0                 2.2.0               No                    This package contains base QoS functionality for the Openswitch software. 

libopx-db-sql1                 1.1.1                 1.1.1               No                    This package contains implementation of database abstractions 

libopx-sai-vm1                 17.1.0                17.1.0              No                    OPX SAI VM - shared library                                 

opx-cps                        3.6.2+opx1            3.6.2+opx1          No                    This package contains the Openswitch object library.  The object library is provides APIs to set/get/delete/create and execute RPC calls using data models provided by components. 
```

**Comment by atanu-mandal** [comment]
I see, this fix is part of opx-nas-interface (5.10.1+opx8)
https://review.openswitch.net/#/c/14583/3/debian/changelog 

It should be part of http://archive.openswitch.net/installers/2.3.0-dev/PKGS_OPX-testing-2.3.0-dev1-installer-x86_64.bin
